src/pybind/mgr/cephadm/module.py

```diff
@@ -1454,8 +1478,7 @@ def _add_host(self, spec):
                 'args': [f'{k}={v}' for k, v in spec.location.items()],
             })
 
-        if spec.hostname not in self.inventory:
-            self.cache.prime_empty_host(spec.hostname)
+        self.cache.prime_empty_host(spec.hostname)
```

Comment:
This is scary either way. I'd actually expect that we're seeing empty hosts here. E.g. because the host was completely replaced. Should't we fetch new host infos and not rely on any existing ones?

Reply:
so we should call prime_empty_host() all the time? or only if the addr changes? if the user is only adding labels to an existing host i would think all the hostcache info would still be valid